runtime/compiler/z/runtime/PicBuilder.m4

```diff
@@ -1961,9 +1961,29 @@ ZZ 31 bit zOS. See definition of J9TR_CAA_SAVE_OFFSET
 
 })dnl
 
-LOAD_ADDR_FROM_TOC(r14,TR_instanceOf)
+ifdef({J9ZOS390},{dnl
+
+LOAD_ADDR_FROM_TOC(r6,TR_instanceOf)
+
+ifdef({TR_HOST_64BIT},{dnl
+
+ZZ 64 bit zOS
+   BASR r7,r6              # call instanceOf
+   LR   r0,r0
```

Comment:
We can probably common up the `BASR r7,r6` as well for the z/OS, as the only difference between 31-bit and 64-bit z/OS would be the size of bytes we need to add as padding after Call for XPLINK.